From 5687040e291c06a7e49cae9cfae0f2be96b05ef3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Sun, 24 Jul 2016 11:15:52 +0200
Subject: [PATCH] Always register perl FTS3 tokenizer with sqlite before 3.11.0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Disabling perl FTS3 tokenizer by default caused t/43_fts3.t failures
if building against sqlite < 3.11.0. That was caused by assymetry
between enabling the support (purely on SQLITE_ENABLE_FTS3_TOKENIZER
environemnt variable) and skipping the tests (on version and
ENABLE_FTS3_TOKENIZER compile option).

CPAN RT#112474

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 SQLite.xs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SQLite.xs b/SQLite.xs
index f20e511..06fc425 100644
--- a/SQLite.xs
+++ b/SQLite.xs
@@ -288,7 +288,7 @@ register_fts3_perl_tokenizer(dbh)
     ALIAS:
         DBD::SQLite::db::sqlite_register_fts3_perl_tokenizer = 1
     CODE:
-#if SQLITE_ENABLE_FTS3_TOKENIZER
+#if SQLITE_VERSION_NUMBER < 3011000 || SQLITE_ENABLE_FTS3_TOKENIZER
         RETVAL = sqlite_db_register_fts3_perl_tokenizer(aTHX_ dbh);
 #else
         RETVAL = 0;
-- 
2.5.5

